Reading 4

<h1>Willows: a memory efficient tree and forest construction package
Heping Zhang*, Minghui Wang and Xiang Chen</h1>


Cited by

<h2>Data mining in the Life Sciences with Random Forest: a walk in the park or lost in the jungle?
Wouter G. Touw, Jumamurat R. Bayjanov, Lex Overmars, Lennart Backus, Jos Boekhorst, Michiel Wels and Sacha A. F. T. van Hijum
 form) : 26th May 2012</h2>

<h3>The four most important keywords:</h3>
<ul>
<li>ii1 Recursive partitioning: A classification rule predicts membership and consists of a predictor and its corresponding threshold. After the splitting rule is applied on the first root, the daughter roots are split and afterwards their daughter rules - recursively.
</li><li>ii2 Deterministic forest: For building a deterministic forest, first consider all competitive splits, and construct the forest with them. The trees are limited on size and have a deterministic and reproducible manner. Compared to random forest it is more efficient to calculate.
</li><li>ii3 Compression Algorithms: All possible choices of SNPs can be represented by 2 bits, so 16 SNPs can be represented by an integer (C++ or Java). Which is more compressed than byte storage or double storage scheme.
</li><li>ii4 single nucleotide polymorphisms(SNPs): variations of a single base pair in a DNA sequence.
</li></ul>

<h3>Brief Description</h3>
<ul><li>iii1 Motivational Statement:
Even though there are existing methods to build powerful tree and forest algorithms, it is difficult to deal with SNP data due to their big size and high memory demand, therefore, they want to build an efficient algorithm that handles this type of bioinformatics data. Their algorithm should be efficient and be feasible to handle those amount of data.
</li><li> iii2 Study instruments:
They use Willows, which is implemented in Java and C++ and runs on Windows, Linux and Mac. It is available under http://c2s2.yale.edu/software/Willows. They used a computer running Windows XP professional and a 2.33 GHz processor with 2 GB physical memory.
</li><li> iii3 Tutorial materials: They explain Classifications Tree, Random Forest, Deterministic Forest and their Compression Algorithm and give references for each of the methods.
</li><li> iii4 Data: They have randomly generated two simulated data sets, one containing 10,000 SNPs and other containing 100,000 SNPs. Both datasets had 1000 subjects. For each of the above datasets, they built classification trees, a random forest containing 100 trees and a deterministic forest containing 8 trees.
</li></ul>

<h3>Improve paper</h3>
<ul><li>iv1 There is no overview, how they use all parts together.
</li><li> iv2 The figures are just briefly explained and they are shown earlier than their explanation.
</li><li> iv3 Everything is explained very briefly and coarse. The algorithm they implemented can be explained more specifically and detailed.
</li><li> iv4 They can explain clearly what their baseline results are to compare their new results with. And how their new method is better than the previous methods.
</li><li> iv5 The best practices in doing these kinds of methods can be explained. The disadvantages of using this methods can be explained

</li></ul>
